src/react/NewPipelinePage.tsx

```diff
@@ -100,7 +125,8 @@ export default ({ astarte, history }) => {
   const createPipeline = useCallback(() => {
     setIsCreatingPipeline(true);
 
-    const pipelineParams = {
+    type PipelineParams = Parameters<typeof astarte.registerPipeline>[0];
```

Comment:
Here I am inferring the Params type since the Client does not directly export a well defined type that specify exactly which params this API method supports. Do you think we should make `astarte-client` export types such as `RegisterPipelineParams` as well?

Reply:
in this specific instance i'd make the API accept a pipeline object, and here would just call `registerPipeline(pipeline)`

Reply:
Sounds right, I modified the API to accept an AstartePipeline